src/solver/optimisation/constraints/ConstraintGroup.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "sim_structure_probleme_economique.h"
+#include "../opt_structure_probleme_a_resoudre.h"
+#include "constraint_builder_utils.h"
+
+class ConstraintGroup
+{
+public:
+    explicit ConstraintGroup(PROBLEME_HEBDO* problemeHebdo, ConstraintBuilder& builder) :
+     problemeHebdo_(problemeHebdo), builder_(builder)
+    {
+    }
+
+    virtual void Build() = 0;
+    PROBLEME_HEBDO* problemeHebdo_;
+    ConstraintBuilder& builder_;
```

Comment:
`ConstraintBuilder& builder_;` : 
**builder_** stored only used to build constraints, but it should not be here, and for 2 reasons :
- As said in **Group1.h**'s review, it should be immediately passed from the group's constructor (any group: **Group1**, ....) to the constructor of constraints the group contains. It shouldn't need to be stored here.
- But **mostly**, it should probably not exist : class **ConstraintBuilder** is too big, and contains logic that constraints need to build themselves. **ConstraintBuilder** should be spread across constraint classes.